code/modules/power/lighting.dm

```diff
@@ -567,13 +567,12 @@
 // if a light is turned off, it won't activate emergency power
 /obj/machinery/light/proc/turned_off()
 	var/area/A = get_area(src)
-	return !A.lightswitch && A.power_light
+	return !A.lightswitch && A.powernet.lighting_powered
```

Comment:
Could you do `!machine_area.lightswitch && machine_area.powernet.lighting_powered` here?

Reply:
done